thunder/dynamo/report.py

```diff
@@ -996,26 +1004,103 @@ def foo(x):
     return report
 
 
-def get_thunder_split_reports(fn: Callable, *args, thunder_compile_kwargs: dict = None, **kwargs):
+def check_torch_runnablility(fn: Callable, *args, **kwargs):
+    try:
+        # WAR for triton error https://github.com/pytorch/pytorch/issues/124565
+        if torch.cuda.is_available():
+            torch.empty(1, device="cuda", requires_grad=True).backward()
+        torch_compiled = torch.compile(fn)
+        run_forward_backward(torch_compiled, *args, **kwargs)
+    except Exception as e:
+        print(f"Failed to run the function using torch.compile with exception: {e}")
+        print(f"Trying with Torch eager...")
+        try:
+            run_forward_backward(fn, *args, **kwargs)
+        except Exception as e:
+            print(f"Failed to run the function with exception: {e}")
+            return
+        print("The input callable can be successfully executed in eager mode.")
+    else:
+        print("The input callable can be successfully executed by torch.compile.")
+
+
+def get_thunder_fxgraph_reports(
+    fn: Callable, *args, thunder_compile_kwargs: dict = None, check_runnablility=True, **kwargs
+) -> list[ThunderFXGraphReport]:
+    """
+    Generates a list of :class:`ThunderFXGraphReport` objects for a given callable.
+
+    This function performs the following steps:
```

Comment:
Would you extend this comment to describe what the arguments are used for, including `*args` and `**kwargs`? 

From a Python UX perspective, I understand why it's convenient to pass args and kwargs like this, but mixing the inputs to the callable and the options to this function can be a confusing user-experience. 

We don't have to change the signature in this PR, but it might make sense to make calling this function look like this:

```
get_thunder_fx_graph_reports(fn, thunder_compile_kwargs=..., check_runnability=...)(*args, **kwargs)
```

which would be done by returning a function or callable class from `get_thunder_fx_graph_reports`. 

fyi @IvanYashchuk as I think he and I discussed the pros and cons of this pattern previously